src/viking.js

```diff
@@ -51,4 +51,90 @@ class Saxon extends Soldier {
 
 
 // War
-class War { }
+class War {
+  vikingArmy = [];
+  saxonArmy = [];
+
+  addViking(vikingObject) {
+    this.vikingArmy.push(vikingObject);
+  }
+
+  addSaxon(saxonObject) {
+    this.saxonArmy.push(saxonObject);
+  }
+
+
+  armyAttack(army) {
+    const randomVikingIndex = Math.floor(
+      Math.random() * this.vikingArmy.length
+    );
+    const randomSaxonIndex = Math.floor(Math.random() * this.saxonArmy.length);
+
+    const randomVikingSoldier = this.vikingArmy[randomVikingIndex];
+    const randomSaxonSoldier = this.saxonArmy[randomSaxonIndex];
+
+    if (army === "Viking") {
+      const attackResult = randomSaxonSoldier.receiveDamage(
+        randomVikingSoldier.strength
+      );
+
+      this.checkDead(randomSaxonSoldier);
+
+      return attackResult;
+    }
+
+    const attackResult = randomVikingSoldier.receiveDamage(
+      randomSaxonSoldier.strength
+    );
+
+    this.checkDead(randomVikingSoldier);
+
+    return attackResult;
+  }
+
+  checkDead(soldier) {
+    if (soldier.health <= 0) {
+      if (soldier instanceof Viking) {
+        const index = this.vikingArmy.indexOf(soldier);
+
+        if (index > -1) {
+          this.vikingArmy.splice(index, 1);
+        }
+      } else {
+        const index = this.saxonArmy.indexOf(soldier);
+
+        if (index > -1) {
+          this.saxonArmy.splice(index, 1);
+        }
+      }
+    }
+  }
+
```

Comment:
Olha só, da linha 76 em diante o código ficou um pouco confuso. Talvez vc tenha pensado numa lógica muito extensa, vou sugerir algo mais curto talvez :)
Como o lab separou vikingAttack e soldierAttack, vou seguir esse raciocínio apenas: da linha 68 ate´ 74 continua nessa lógica, tanto para o método vikingAttack como soldierAttack, mudando apenas variáveis de cálculo de acordo com viking ou saxon. 
Feito isso, vamos para a lógica da luta, que vc separou em cheackdead e atttackResult, podendo ser apenas uma linha de checagem para checar o seu 'checkdead' assim: 
let fight = randomSaxonSoldier.receiveDamage(randomVikingSoldier.attack( ))
E aqui um if para ver a health do saxon no caso: 
 if(randomSaxonSoldier.health <= 0){
            this.saxonArmy.splice(randomSaxonIndex, 1)
        }
        return fight
}
Usamos várias informações do código todo, de maneira mais resumida. 
Depois de fazer para o vikingAttack, faz a mesma lógica (só que reversa entre viking e saxon) para saxonAttack 
Veja se entendeu e me fala :)